### PR TITLE
Adds Arcade Tokens and changes Delta's maint arcade appropriately

### PR DESCRIPTION
**Pull request by Kerbin-Fiber**
## About The Pull Request
Arcade Machines now reward you with Arcade Tokens instead of giving you prizes.
This doesn't remove the prizes, just use the arcade token on the machine to receive a prize!
Also makes bananium coins give you prizes when put into an arcade machine. (Because spending a valuable rare resource that's worth 200 credits on a toy is completely reasonable.)

DeltaStation's maintenance arcade coins were broken, as their sprite was deleted and were still using them. Making them default to iron. This changes those coins to some free tokens to get prizes.

Disclaimer: You are not able to stash an arcade token, emag an arcade game and instert token for free bomb. You still have to beat the game properly to win your devilish prize. That prize will still immediately come out it won, no tokens.

## Why It's Good For The Game
Muh immersion. Also a step closer to making a toy vendor that give you Toy Esword for something like 10 arcade tokens.

## Changelog
:cl:
add: Added Arcade Tokens! Redeem prizes from arcade machines with these!
add: Bananium coins work like Arcade Tokens when put into an arcade machine!
del: Removed DeltaStation's broken arcade coins in place of Arcade Tokens!
tweak: Arcade machines no longer give you your prizes instantly! They give you arcade tokens, just put the token in the machine to receive your prize! Get your moneybags ready!
/:cl:

### CODE REVIEW

**Comment by Kerbin-Fiber** [comment]
fuck you mapmerge

**Comment by ArcaneMusic** [comment]
Good change, I can use these when I get around to porting the Arcade Game to the Modular Computer.

File: code/game/machinery/computer/arcade.dm [hunk 1]
```diff
@@ -79,19 +80,33 @@ GLOBAL_LIST_INIT(arcade_prize_pool, list(
...
 	visible_message("<span class='notice'>[src] dispenses [the_prize]!</span>", "<span class='notice'>You hear a chime and a clunk.</span>")
-
 	the_prize.forceMove(get_turf(src))
+	
+/obj/machinery/computer/arcade/attackby(obj/item/W, mob/user)
+	if(istype(W, /obj/item/coin/arcade_token) || istype(W, /obj/item/coin/bananium))
+		to_chat(user, "<span class='notice'>You insert the [W] into the [src].")
+		redeem(user)
+		qdel(W)
+		return
```
Comment (nemvar) [inline code/game/machinery/computer/arcade.dm]: return TRUE here.

Comment (Kerbin-Fiber) [reply code/game/machinery/computer/arcade.dm]: donezo

**Comment by Qustinnus** [comment]
Don't actually make it so you can select what items you get, the rng of prizes is part of what, makes it fun. People buying spac tiles every round is just going to ruin them for example.

**Comment by Kerbin-Fiber** [comment]
Well then, I really need to learn how buckling works so I can start on carp saddles instead.

**Comment by Ty-the-Smonk** [comment]
Local xenobiologist gets 5 stacks of bannanium and gets 12 pulse rifles

File: code/game/machinery/computer/arcade.dm [hunk 2]
```diff
@@ -1090,7 +1105,9 @@ GLOBAL_LIST_INIT(arcade_prize_pool, list(
 	gameStatus = ORION_STATUS_START
 	say("Congratulations, you made it to Orion!")
 	if(obj_flags & EMAGGED)
-		new /obj/item/orion_ship(loc)
+		prizeselect = /obj/item/orion_ship
+		prizevend(user)
+		prizeselect = /obj/item/coin/arcade_token
```
Comment (ATH1909) [inline code/game/machinery/computer/arcade.dm]: what about the other emagged arcade machine prize from the other kind of emagged arcade machine?

Comment (Kerbin-Fiber) [reply code/game/machinery/computer/arcade.dm]: The Arcade Battle game checks if you win during every action, and in doing so, it dispenses the bomb and the rare sombrero in an emag check. So it doesn't use the prizevend proc to give out it's emagged prize

**Comment by Kerbin-Fiber** [comment]
> Local xenobiologist gets 5 stacks of bannanium and gets 12 pulse rifles

Bruh
> Disclaimer: You are not able to stash an arcade token, emag an arcade game and instert token for free bomb. You still have to beat the game properly to win your devilish prize. On top of that, you aren't able to throw in a million tokens and expect the pulse rifle to come out.

**Comment by Kerbin-Fiber** [comment]
Wow, fuck me I guess

File: code/game/machinery/computer/arcade.dm [hunk 3]
```diff
@@ -80,20 +81,36 @@ GLOBAL_LIST_INIT(arcade_prize_pool, list(
 		new /obj/item/gun/energy/pulse/prize(src)
 		visible_message("<span class='notice'>[src] dispenses.. woah, a gun! Way past cool.</span>", "<span class='notice'>You hear a chime and a shot.</span>")
 		SSmedals.UnlockMedal(MEDAL_PULSE, user.client)
+    
+	else
+  
+		new prizeselect(src)
+
```
Comment (actioninja) [inline code/game/machinery/computer/arcade.dm]: remove the wonky extra gaps

File: code/game/machinery/computer/arcade.dm [hunk 4]
```diff
@@ -80,20 +81,36 @@ GLOBAL_LIST_INIT(arcade_prize_pool, list(
...
+	var/atom/movable/the_prize = pick(contents)
+		visible_message("<span class='notice'>[src] dispenses [the_prize]!</span>", "<span class='notice'>You hear a chime and a clunk.</span>")
 		return
 
+	the_prize.forceMove(get_turf(src))
+
```
Comment (actioninja) [inline code/game/machinery/computer/arcade.dm]: this is error because the indentation is fucked. Also you have a return before the last line of the proc.

**Review by actioninja — CHANGES_REQUESTED** [review]

File: code/game/machinery/computer/arcade.dm (hunk 4 above)
Comment (Kerbin-Fiber) [reply code/game/machinery/computer/arcade.dm]: ah

**Comment by Kerbin-Fiber** [comment]
I think that #46820 kinda fucked up the code cuz uhh, on my side, those indents aren't there, and git is telling me that this code and my code are the same, but they're clearly not.

**Comment by Kerbin-Fiber** [comment]
lemme kill git and fix it

**Comment by actioninja** [comment]
Stale with no changes for 10 days.
Feel free to ask for a reopen or make a new PR if you fix the issues.

**Pull request closed without merging**